app/controllers/api/connect/v3/subscriptions/systems_controller.rb

```diff
@@ -1,10 +1,19 @@
 class Api::Connect::V3::Subscriptions::SystemsController < Api::Connect::BaseController
 
   def announce_system
-    system = System.create!(hostname: params[:hostname]) # TODO: store hwinfo data
+    system = System.create!(
+      hostname: params[:hostname],
+      hw_info_attributes: hwinfo_params
+    )
 
     logger.info("System '#{system.hostname}' announced")
     respond_with(system, serializer: ::V3::SystemSerializer, location: nil)
   end
 
+  private
+
+  def hwinfo_params
```

Comment:
Please `hw_info_params` instead of `hwinfo_params` if you already set the standard name in the database.